### PR TITLE
Add config option to confirm kicks from Clan Chat

### LINKED ISSUE #727: Add clan chat join alerts, something to avoid accidental kicks
- Join alerts on: total level, P2P/F2P exp, F2P/P2P/RS3/RS3Lobby worlds
- Chat message right click -> Kick from CC
- Something to  avoid accidental kicks, maybe to do with ordering of the cc members list?

### PR DESCRIPTION
**Pull request by TheStonedTurtle**
Adds a new config option to the ClanChat plugin that defaults to on.

vE9YXpoo2o

closes #727

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/ScriptID.java [hunk 1]
```diff
@@ -194,4 +194,13 @@
 	 */
 	@ScriptArguments(integer = 2)
 	public static final int XPDROP_DISABLED = 2091;
+
+	/**
+	 * Attempts to kick the specified player from the Clan Chat
+	 * <ul>
+	 * <li> String Players in-game name
```
Comment (Adam-) [inline runelite-api/src/main/java/net/runelite/api/ScriptID.java]:
```suggestion
	 * <li>String Players in-game name</li>
```

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatConfig.java [hunk 2]
```diff
@@ -137,4 +137,15 @@ default boolean clanTabChat()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+		keyName = "confirmKicks",
+		name = "Confirm Kicks",
+		description = "Shows a chat prompt to confirm the kick before kicking them",
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatConfig.java]:
```suggestion
		description = "Shows a chat prompt to confirm kicks",
```

File: runelite-client/src/main/scripts/ClanSendKick.rs2asm [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+   sconst                 "-Attempting to kick player from friends chat..."
+   iconst                 2
+   invoke                 96
+   sload                  0
+   clan_kickuser
+   jump                   LABEL73
+LABEL73:
+   return
+CONFIRM_KICK:
+   sload                  0             ; Username we are trying to kick
```
Comment (Adam-) [inline runelite-client/src/main/scripts/ClanSendKick.rs2asm]: this needs to be popped too

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 4]
```diff
@@ -623,4 +654,18 @@ private void addClanCounter()
...
 	}
+
+	private void confirmKickPlayer(final String kickPlayerName)
+	{
+		chatboxPanelManager.openTextMenuInput("Attempting to kick: " + kickPlayerName)
+			.option("1. Confirm kick", () ->
+				clientThread.invoke(() ->
+				{
+					kickConfirmed = true;
+					client.runScript(ScriptID.CLAN_SEND_KICK, kickPlayerName);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: You could just move the kick confirmed unset to here

**Comment by TheStonedTurtle** [comment]
The PR has been updated as requested and is ready for re-review

**Comment by Adam-** [comment]
This seems good, though I would turn it off by default

**Pull request merged**